tools/cli/src/mentat_cli/input.rs

```diff
@@ -71,6 +78,44 @@ impl InputReader {
         self.reader.is_some()
     }
 
+    pub fn read_file(&mut self, filename: &String) -> Result<Vec<InputResult>, cli::Error> {
+        let path = Path::new(&filename);
+        let display = path.display();
+
+        let f = match File::open(&path) {
+            Err(err) => bail!(cli::ErrorKind::FileError(display.to_string(), err.to_string())),
```

Comment:
You define `FileError`, so make it own a `Path` and a `std::io::Error`, and don't convert to strings everywhere!